### ISSUE TITLE
Maven build fails due to missing javafx dependency

### ISSUE THREAD

**Issue by wchargin** [westoncb/HNDesktop#10]
Fresh Linux Mint 18.2 (Ubuntu 16.04) install, with `apt install maven` followed by `mvn package`, yields a bunch of lines of output and then
```
[INFO] Compiling Kotlin sources from [/home/wchargin/git/HNDesktop/src/main/java]
[INFO] Module name is HNDesktop
[ERROR] /home/wchargin/git/HNDesktop/src/main/java/App.kt: (5, 8) Unresolved reference: javafx
[ERROR] /home/wchargin/git/HNDesktop/src/main/java/App.kt: (6, 8) Unresolved reference: javafx
[ERROR] /home/wchargin/git/HNDesktop/src/main/java/App.kt: (7, 8) Unresolved reference: javafx
[ERROR] /home/wchargin/git/HNDesktop/src/main/java/App.kt: (8, 8) Unresolved reference: javafx
[ERROR] /home/wchargin/git/HNDesktop/src/main/java/App.kt: (72, 20) Unresolved reference: JFXPanel
[ERROR] /home/wchargin/git/HNDesktop/src/main/java/App.kt: (223, 13) Unresolved reference: Platform
[ERROR] /home/wchargin/git/HNDesktop/src/main/java/App.kt: (224, 31) Unresolved reference: WebView
[ERROR] /home/wchargin/git/HNDesktop/src/main/java/App.kt: (225, 34) Unresolved reference: Scene
```

I haven't used JavaFX or Kotlin, but I see that it's not mentioned in the `pom.xml` anywhere. Perhaps you need to add a dependency?

Maven and Java versions:
```
$ mvn -version
Apache Maven 3.3.9
Maven home: /usr/share/maven
Java version: 1.8.0_131, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-8-openjdk-amd64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.10.0-32-generic", arch: "amd64", family: "unix"
```

**Comment by westoncb** [comment]
Hey thanks for pointing this out wchargin! AFAIK JavaFX is generally included as part of the standard library, but maybe there are certain cases where it's not included... In any case, I did find a javafx maven plugin which seems like it could be a solution. If you'd like to try it out, you could pull the latest on the branch `javafx-fix` and run `mvn package` again and see what happens.

**Comment by wchargin** [comment]
Are you sure that you pushed? I am "Already up-to-date" on 8d473ab8c9, which is also the latest GitHub commit and was created before your comment. I don't see any other branches on the remote.

Try `git push origin javafx-fix`?

**Comment by westoncb** [comment]
Haha! Yeah, pushing might be useful—thanks again @wchargin

**Comment by wchargin** [comment]
The commit currently at `javafx-fix`, ebfacff7f2, does not fix the problem for me. But I suspect that you might not have pushed the right thing, as ebfacff7f2 is the _parent_ commit of 8d473ab8c9, the current master commit.

If your HEAD is at the commit that you think fixes the problem, you can `git push origin HEAD:javafx-fix`, or `git push --force-with-lease origin HEAD:javafx-fix` if the preceding command has a "non-fastforward" error.

**Comment by westoncb** [comment]
Oh wow, yep—I was working in another copy of the repo I had cloned in order to test building out. Correct branch should be pushed now.

**Comment by wchargin** [comment]
Cool—I now have what appears to be the correct commit, 1fd7df4d11. Unfortunately, I get the same error as before when running `mvn package`. I also tried running `mvn jfx:jar`, which has the same error.

Any other suggestions?

**Comment by westoncb** [comment]
I noticed your `mvn -version` lists Oracle as the Vendor, but your `Java home` is point to an Open JDK, which apparently does not come with JavaFX. I found this: https://stackoverflow.com/questions/34243982/why-is-javafx-is-not-included-in-openjdk-8-on-ubuntu-wily-15-10 —which suggests using `sudo apt-get install openjfx` to patch your jdk.

It may be that you've got to do that, or use an Oracle jdk.
